### ISSUE TITLE
Make relationship between signals and hamiltonian_operators explicit in tutorial.

### ISSUE THREAD

**Issue by hodgestar** [Qiskit-Extensions/qiskit-dynamics#274]
In https://qiskit.org/ecosystem/dynamics/tutorials/Rabi_oscillations.html the relationship between the `hamiltonian_operators`, the `signals` passed to the solver and the time-dependent term in the Hamiltonian isn't explicitly stated. I think readers who are not familiar with how such solvers structure things might not easily pick this up.

I also find the name `hamiltonian_operators` for the time-dependent / signal operators a bit strange. The static part of the Hamiltonian is also a Hamiltonian operator. Perhaps naming the argument `signal_operators` might this clearer?

**Comment by DanPuzzuoli** [comment]
I agree the naming is a little odd. It's been a while so I can't remember the alternatives we had considered, but I think we settled on the current naming conventions as a compromise between clarity and wanting to avoid the argument names being too long. (E.g. `signal_operators` isn't quite correct either as there are signals that can be supplied as coefficients for dissipator terms as well.)

At the stage I'm resistant to changing them as they've been with the package for so long. However, I think giving more explanation in the tutorial would be good. I've also just reread the `Solver` class documentation and that can be clarified as well.

**Comment by Naohnakazawa** [comment]
Hi @DanPuzzuoli, may I work on this isuue?

**Comment by DanPuzzuoli** [comment]
Hi @Naohnakazawa - of course :)

**Comment by Naohnakazawa** [comment]
Hello @hodgestar, I have opened a PR [#388](https://github.com/qiskit-community/qiskit-dynamics/pull/388) for this issue fix.
Could you approve?

**Comment by hodgestar** [comment]
@Naohnakazawa I don't have approval rights for this repository.
